### PR TITLE
March 2017 meeting notes

### PR DESCRIPTION
**Pull request by leobalter**
the folder structure is weird, but I'm trying to capture the essence of it.

I already changed my initials to avoid conflict with a pre-existing LB (Lee Byron)

I'll open another PR to change my LB initials to LBR as well.

### CODE REVIEW

File: es8/2017-03/mar-21.md [hunk 1]
```diff
@@ -0,0 +1,1457 @@
...
+LBR: Yes
+AWB: So get the link up there.
+
+## Any items for the 2018 edition?
+
+AWB: we talked about the 2017 edition of the standard. Are there any items for the 2018 edition?
+
+BT: One of the big things I've been pushing this year (and plan to push in a big way next year) is machine readability. Specifically reducing the number of terms used in ECMASpeak. Specifically, I'm looking into making ECMAIDL/a subset of WebIDL.
+
+YK: We should really be willing to use a webset of WebIDL. There may be something more suitable for webspecs that's more suitable for JS semanatics.
```
Comment (hax) [inline es8/2017-03/mar-21.md]: subset of WebIDL

File: es8/2017-03/mar-21.md [hunk 2]
```diff
@@ -0,0 +1,1457 @@
...
+
+BT: This is not at all concerning. This is a well understood feature that exists in other regex implementations.
+
+AWB: In the absence of objections, we should consider advancing this to stage 3
+
+YK: I think V8 is happy with the ??? I assume V8 is ok with those changes.
+
+MF:  I have gotten feedback from Kevin, he said it looks good. The only issues were typos.
+
+BT: So Dan, do you want to talk more wholistically about what you think the state of RegEx will be for 2018
```
Comment (hax) [inline es8/2017-03/mar-21.md]: holistically

File: es8/2017-03/mar-21.md [hunk 3]
```diff
@@ -0,0 +1,1457 @@
...
+
+DE: At the last meeting, Allen said, for ES2018 wouldn't it be nice if we could send a message that shows all the nice features for RegExp. We'd have everyone's work together. We have look behind, ...., and named groups, dotall flag, and match all. So, for match all, that's currently at stage-1. I think maybe we can get it through by ES2018. I was wondering if it'd make sense to go with semantics that are analogous to the current regular expression thing similar to split where it ends up cloining the RegExp. On the other hand, we've previously discussed adding a Symbol.exec method for being the basis for subclassing RegExps. I'm more hesitant about that. But I think we could do matchAll separately from this. So I started talking with Jordan about this and hope we can get all 5 of these things through by 2018 to have a coherent story for RegExp.
```
Comment (hax) [inline es8/2017-03/mar-21.md]: `cloining` ?

File: es8/2017-03/mar-22.md [hunk 4]
```diff
@@ -0,0 +1,773 @@
...
+
+MM: Particulars: The placement indicator is leading keyword. 
+
+MM: The presence of a leading static keyword means "put it on the constructor object." The presence of an initial "own" keyword means "put it on the instance object." I've gotten push-back on the "own keyword, but it needs ot not be "private or "public" since that's the kind of sematics we are trying to break up into these combinations.
+
+MS: Is static own a syntax error?
+
+MM: Yes
+
+YK: Is that an orthagonality violation? (making speciic rules about "static" and "own")
```
Comment (hax) [inline es8/2017-03/mar-22.md]: orthogonality

File: es8/2017-03/mar-21.md [hunk 5]
```diff
@@ -0,0 +1,1457 @@
...
+
+WH: Whichever you do, some people will want the other, so do what makes sense.
+
+YK: Work together with the Duration proposal.
+
+CP: Let's keep the proposals separate; it's easy for me to integrate them separately.
+
+ZB: So ListFormat goes to Stage 2 and the others stay at Stage 1
+
+ZB: The last proposal I have is a proposal that I'm seeking stage 1 for. I presented it at the last meeting. It's a proposal for simplifying the way of formatting date and times using two formats `dateStyle` and `timeStyle` instead of doing them separately, we would actually like to use one of the four lengths of style `"short", "medium", "long" and "full". I think it's a pretty good representation. It makes it easier to use DateTimeFormat, but it also allows one more thing. It is language independent. For example, the "short date" specifiers are relative to the current locale. By using dateStyle or timeStyle the user hands of control of the formatting to the language. We expect this will be the most common way people use DateTimeFormat.
```
Comment (azu) [inline es8/2017-03/mar-21.md]: Is it(\`) unnecessary?

tc39-notes 2017-04-02 21-51-30

File: es8/2017-03/mar-22.md [hunk 6]
```diff
@@ -0,0 +1,773 @@
...
+
+AWB: There's actually a pull request that Mark hasn't accepted yet, the paragraph on constructors is kind of bogus. Has a whole file of examples taken from some of the existing proposals recast.
+
+MM: Wonderful
+
+MM: Any objection?
+
+BT: I have a number of points, but let's let Dan go first
+
+DE: My intuition is that this is one more dimension than what might be the common understanding of objects.  This dimension of "own" vs. being on the prototype is definitely there, but the way that we look at classes, e don't have to think about adding the method ot the prototype.  With the public and private fields proposal, and the private methods straw-man, it sort of retains this property.  You do't have to be explicit about the prototype vs. the instance.  This is intuitive and fuzzy but seems important. Talking about "woned" vs. "notowned" gets into teh weeds and will confuse people.  I'm espcially concerned about non/owned methods.  People might think they are supposed ot put "owned" on everything, which woul dbe hard to optimize.
```
Comment (hax) [inline es8/2017-03/mar-22.md]: "woned" -> "owned"
"woul dbe" -> "would be"

Comment (ljharb) [inline es8/2017-03/mar-22.md]: teh weeds -> the weeds; ot -> to; e don't -> we don't; do't -> don't

File: es8/2017-03/mar-21.md [hunk 7]
```diff
@@ -0,0 +1,1457 @@
...
+
+AK: it's nice to have it advance if it could. But what we got out of it is that it needs eyes on it.
+
+MM: withdrawn from stage 3 for now.  
+
+AWB: are you planning to present it at the next meeting?
+
+MM: yes.
+
+DH: Add it to the agenda for next
```
Comment (kgryte) [inline es8/2017-03/mar-21.md]: I very much agree with this. This is especially true for numeric computing use cases where scripting is done in JavaScript. Having to reach for the entire WASM toolchain would be __very__ burdensome.

**Comment by rwaldron** [comment]
> the folder structure is weird, but I'm trying to capture the essence of it.

I tried to change this about a year ago (ie. remove the version part of the directory structure), but when I reached out to @ForbesLindesay w/r to updating https://esdiscuss.org/notes, I believe he was super busy at the time and we casually agreed to put it off for the time being. Looking at the PRs for that project, there is a patch to flatten dirs, but the author never responded to a request for a minor change: https://github.com/esdiscuss/esdiscuss.org/pull/138 I bet @ForbesLindesay would be glad to accept a corrected version of that PR.

File: es8/2017-03/mar-21.md [hunk 8]
```diff
@@ -0,0 +1,1457 @@
+# March 21, 2017 Meeting Notes
+
+Allen Wirfs-Brock (AWB), Waldemar Horwat (WH), Brian Terlson (BT), Michael Ficarra (MF), Adam Klein (AK), Dave Herman (DH),  Kent C. Dodds (KCD), Tim Disney (TD), Daniel Ehrenberg (DE), Shu-yu Guo (SYG), Michael Saboff (MS), Sebastian Markbage (SM), Bradley Farias (BF), Maggie Pint (MPT), Jamund Ferguson (JF), Myles Borins (MBS), Logan Smyth (LS), Sarah D'Onofrio (SD), Alan Schmitt (AS), Dean Tribble (DT), Peter Jensen (PJ), Mark S. Miller (MM), Leo Balter (LBR), Zibi Braniecki (ZB), Rafael Xavier (RX), Yehuda Katz (YK), Caridy Patino (CP), Diego Ferreiro Val (DFV), Brendan Eich (BE), Lyza Gardner (LG), Mathias Bynens (MB)
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: - Logan Smyth's LS -> LSH
- Sarah D'Onofrio's SD -> SDO
- Lyza Gardner's LG -> LGR
- Jamund Ferguson's JF -> JFN
- Rafael Xavier's RX -> RXR

Comment (rwaldron) [inline es8/2017-03/mar-21.md]: (if I missed any other new members, please just change them)

File: es8/2017-03/mar-21.md [hunk 9]
```diff
@@ -0,0 +1,1457 @@
...
+
+## 3. Adopting the agenda
+
+DT: WeakReferences. Please add 2 additional minutes.
+
+AK: Please add items 1 week in advance.
+
+Agenda adopted
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Missing link to agenda

File: es8/2017-03/mar-21.md [hunk 10]
```diff
@@ -0,0 +1,1457 @@
...
+
+AWB: the patent document is out for review. Members should get it to their lawyers.
+
+## 7. ECMA402 Status Updates
+
+(Zibi Braniecki)
+
+ZB: I'm not prepared for this discussion. I believe Caridy will be giving this. Will come back in a few minutes
+
+6.ii Vote to accept ECMA-262 2017 final draft and forward it to Ecma GA (approval subject to completion of RF opt-out period with no opt-outs)
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: This agenda item needs to be a header:

```
## 6.ii Vote to accept ECMA-262 2017 final draft...
```

File: es8/2017-03/mar-21.md [hunk 11]
```diff
@@ -0,0 +1,1457 @@
...
+
+AWB: I suggest we do a single vote covering both docs.
+
+LBR: We also have the ECMA TR104.
+
+AWB: We will handle that separately.
+
+AWB: Let's go around the room and have those who have voting authority make your vote. Note takers please take not of the organization who is voting.
+
+Tilde: yes
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Was this YK?

File: es8/2017-03/mar-21.md [hunk 12]
```diff
@@ -0,0 +1,1457 @@
...
+
+DH: Mozilla Yes
+
+WH: Google yes
+
+Shape security: Yes
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Who?

File: es8/2017-03/mar-21.md [hunk 13]
```diff
@@ -0,0 +1,1457 @@
...
+
+FB: Yes
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Who?

File: es8/2017-03/mar-21.md [hunk 14]
```diff
@@ -0,0 +1,1457 @@
...
+
+MS: Apple: Yes
+
+GoDaddy: Yes
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Who?

File: es8/2017-03/mar-21.md [hunk 15]
```diff
@@ -0,0 +1,1457 @@
...
+
+BT: Microsoft: Yes
+
+JSFoundation: Yes
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Who?

File: es8/2017-03/mar-21.md [hunk 16]
```diff
@@ -0,0 +1,1457 @@
...
+
+Intel: Yes
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Who?

File: es8/2017-03/mar-21.md [hunk 17]
```diff
@@ -0,0 +1,1457 @@
...
+
+INRIA: Yes
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Who?

File: es8/2017-03/mar-21.md [hunk 18]
```diff
@@ -0,0 +1,1457 @@
...
+
+KD: PayPal: Yes
+
+DFV: SalesForce: Yes
+
+AWB: Someone will count them. We got "yesses" and no "no"s.  It will be referred to General Assembly for approval. Our work is done for last year. 
+
+All: _clapping_
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: There should be a complete conclusion/resolution section here that includes the resolution to submit the spec for GA approval.

```
#### Conclusion/Resolution

- ....
```

File: es8/2017-03/mar-21.md [hunk 19]
```diff
@@ -0,0 +1,1457 @@
...
+
+8. Test262 Status Updates
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Missing header


```
## 8. Test262 Status Updates 
```

File: es8/2017-03/mar-21.md [hunk 20]
```diff
@@ -0,0 +1,1457 @@
...
+
+BT: I'm coming at it from an implementers standpoint.
+
+MM: We should just have a face-to-face conversation Brian to talk it out.
+
+BT: Ok.
+
+SYG: I have been told Caridy is on the call now
+
+#### Resolution/Conclusion
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]:
```
#### Conclusion/Resolution
```

File: es8/2017-03/mar-21.md [hunk 21]
```diff
@@ -0,0 +1,1457 @@
...
+
+ZB: We have an Intl formatter which is Intl.NumberFormat. And you can also override `toLocaleString`. This formatter could also be the same for arrays
+
+WH: What will it do on an empty array?
+
+ZB: Return an empty string (like it does now)
+
+AWB: The fact that is extending `toLocaleString` in places *should* be noted, but ... I guess I'm interested in the background of this feature in terms of its usefulness. Are there other internationzliation packages that have this thing. 
+
+CP(??): Yeah, I can speak about that. We do have prior art. Java has an API for it. In terms of the user... You're seeing this in almost every social network when you have a list of people who like your tweet/facebook post, you have to list them based on the locale of the user. But we're saying for all your formatters, we need to have data to reliably list an array, the user would have to do this themselves. This just aligns with the existing formatters.
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Need to remove the "(??)" part

File: es8/2017-03/mar-21.md [hunk 22]
```diff
@@ -0,0 +1,1457 @@
...
+
+JF: I know the folks at PayPal were working on something similar and would probably like to review this.
+
+ZB: It really depends on ...???? They are separate spec proposals. I don't think the spec is finalized for them. How about we have ListFormat advance to stage 2 and get interest on the others later.
+
+AWB: Yes, let's do that.
+
+ZB: So PayPal said they may be interested in helping review this?
+
+JF: Yeah, I think we should be able to help out.
```
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: No Conclusion here?

File: es8/2017-03/mar-21.md (hunk 5 above)
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Looks like it just needs to be closed at the end of the word "full"

File: es8/2017-03/mar-22.md [hunk 23]
```diff
@@ -0,0 +1,775 @@
...
+
+AWB: I do, too, but I want to be fair. This is one of those things that Ecma as an organization: they think it's good for them.
+
+#### Conclusion/Resolution
+
+- Consensus reached to approve and forward to Ecma GA
+
+## 12. i. Needs-Consensus PRs
+
+### #856 Avoid Duplicate TA#fill value coercions
```
Comment (rwaldron) [inline es8/2017-03/mar-22.md]: Make this a link to the PR

File: es8/2017-03/mar-22.md [hunk 24]
```diff
@@ -0,0 +1,775 @@
...
+
+AWB: No, but I do think you should duplicate the whole algorithm.
+
+BT: It is duplicated.
+
+#### Conclusion/Resolution
+
+- consensus
+
+### #854 Remove Implementation-defined behavior for Number methods
```
Comment (rwaldron) [inline es8/2017-03/mar-22.md]: Make this a link to the PR

File: es8/2017-03/mar-22.md [hunk 25]
```diff
@@ -0,0 +1,775 @@
...
+
+YK: I had an issue tht was similar to this last week. I had a language binding and I had to decide about what a float in langauge A would be in language B.
+
+AWB: I think we should take this input as input to a future proposal or PR.
+
+#### Conclusion/Resolution
+
+- consensus
+
+### #785 Array.prototoype.sort checks for compareFn being valid
```
Comment (rwaldron) [inline es8/2017-03/mar-22.md]: Make this a link to the PR

File: es8/2017-03/mar-22.md [hunk 26]
```diff
@@ -0,0 +1,775 @@
...
+
+BT: That change was fairly recent too, not calling the compareFn if only one element.
+
+BT: Any objections to taking this pull request?
+
+#### Conclusion/Resolution
+
+- consensus
+
+### #833 Remove duplicate keys from proxies
```
Comment (rwaldron) [inline es8/2017-03/mar-22.md]: Make this a link to the PR

File: es8/2017-03/mar-22.md [hunk 27]
```diff
@@ -0,0 +1,775 @@
...
+
+AWB: ... symbols?
+
+AK: As the editor, keep an eye on that. I fear that someone will use "duplicate" somewhere and not know what it means.
+
+#### Conclusion/Resolution
+
+- consensus
+
+### #852 Move side-effecty toIndex in TA constructor to before detached check
```
Comment (rwaldron) [inline es8/2017-03/mar-22.md]: Make this a link to the PR

File: es8/2017-03/mar-22.md [hunk 28]
```diff
@@ -0,0 +1,775 @@
...
+
+AWB: Does moving it up change the order in which arguments are examined?
+
+BT: It does not change the observable order of side effects. It should be fine to do and not cause compat problems.
+
+#### Conclusion/Resolution
+
+- consensus
+
+### #853 String.prototype.replace edge case alignment
```
Comment (rwaldron) [inline es8/2017-03/mar-22.md]: Make this a link to the PR

File: es8/2017-03/mar-22.md [hunk 29]
```diff
@@ -0,0 +1,775 @@
...
+
+BT: I will add notes with history that Waldemar provided. 
+
+MPT: Did you want to do the needs consensus on ??
+
+BT: I'm open to continuing. 
+
+#### Conclusion/Resolution
+
+- consensus
```
Comment (rwaldron) [inline es8/2017-03/mar-22.md]: What is the consensus for?

**Comment by leobalter** [comment]
I imported the fixes from #65 and applied other final fixes at ee8762d.

This should be ready to land.

@rwaldron While we don't have that fix in esdiscuss, is there anything we can do better for now?

File: es8/2017-03/mar-23.md [hunk 30]
```diff
@@ -0,0 +1,1513 @@
...
+
+WH: There are really two obvious choices. There are only two I can think of that do reductions of an Integer x modulo 2⁶⁴: `asUintN(64, x)` and `x & 0xFFFFFFFFFFFFFFFFn`.
+
+DH: Those get you into the type. You also have to know what operations you can do to keep you in the type.
+
+WH: It's the same ones.
+
+DH: Right, but in order to do arithmetic with the type, you have to understand how to compose this.
+
+WH: Integers form a mathematical ring. Uint64's form a finite ring. The ring of mathematical integers maps nicely to the ring of integers modulo 2⁶⁴ in the obvious way. So you can do as many of the ring computations (+, -, *) as you want using mathematical integers and do one reduction modulo 2⁶⁴ at the end, and you'll always get the same result as if you had been using Uint64 modulo arithmetic throughout.
```
Comment (rwaldron) [inline es8/2017-03/mar-23.md]: > (+, -, *)

The `*` needs to be escaped, like: `\*`

File: es8/2017-03/mar-23.md [hunk 31]
```diff
@@ -0,0 +1,1513 @@
...
+
+WH: The current proposal is that 0 is falsey and all other numbers are truthy.
+
+BE: There's precedence in the committee going back to decimal.
+
+MM: How do you imagine 0 falsy extending to user-defined types?
+
+BE: Value types would have an examplar 0 value.
+
+YK: I have a question about what else we might do here. There is a place in
```
Comment (rwaldron) [inline es8/2017-03/mar-23.md]:
```
#### Conclusion/Resolution
```

File: es8/2017-03/mar-21.md (hunk 8 above)
Comment (leobalter) [inline es8/2017-03/mar-21.md]: I found no other people assigned to LS, SD, LG, JF, or RX.

I did searches on the GitHub repo and also locally using many git grep searches.

examples:

```
git grep RX **/*.md
git grep -i 'rx:'
```

Comment (leobalter) [inline es8/2017-03/mar-21.md]: AFAIK, the rule is: initials for first name and last name, if it's taking, append the last letter in the last lane.

File: es8/2017-03/mar-21.md (hunk 22 above)
Comment (leobalter) [inline es8/2017-03/mar-21.md]: Actually no. The next is a `###` sublink for `## 10.i.f ECMA-402 Presentation of the three formatters for stage 2`

The presentation was long.

**Comment by leobalter** [comment]
@rwaldron fixes applied

File: es8/2017-03/mar-21.md (hunk 22 above)
Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Ok

**Comment by littledan** [comment]
Summary:

1. A [Code of Conduct](https://tc39.github.io/code-of-conduct-proposal/) was presented, based on an open process of feedback over the previous two months. The committee shares the goal of being open and welcoming to diversity and is working through issues of scope and enforcement. The current draft will be presented to the ECMA Secretariat for further review.
1. There is an [alternative proposed unifying syntax](github.com/erights/Orthogonal-Classes) for class features.
1. [Several](https://github.com/tc39/proposal-regexp-lookbehind) [RegExp](https://github.com/tc39/proposal-regexp-unicode-property-escapes) [feature](https://github.com/tc39/proposal-regexp-named-groups) [enhancements](https://github.com/mathiasbynens/es-regexp-dotall-flag) reached Stage 3; these [and](https://github.com/tc39/proposal-regexp-legacy-features) [more](https://github.com/tc39/String.prototype.matchAll) may be in store for ES2018
1. Arbitrary-precision [Integers](https://github.com/tc39/proposal-integer) were introduced and reached Stage 2.
1. A [new built-in Date replacement](https://github.com/maggiepint/proposal-temporal) reached Stage 1.
1. The [Stage 1 Realms proposal](https://github.com/caridy/proposal-realms) has been solidifying, with hooks for the module loading pipeline.
1. [Several](https://github.com/tc39/proposal-intl-segmenter) [Intl](https://github.com/zbraniecki/proposal-intl-unit-format) [proposals](https://github.com/zbraniecki/intl-relative-time-spec) were presented; [Intl.ListFormat](https://github.com/zbraniecki/intl-list-format-spec) reached Stage 2 and [dateStyle/timeStyle](https://github.com/zbraniecki/proposal-ecma402-datetime-style) reached Stage 1.
1. [WeakRefs](https://github.com/tc39/proposal-weakrefs) is seeking implementations to work with and develop feedback.
1. Browsers are no longer pursuing SIMD.js, in favor of SIMD exposed by WASM.
1. The role of Annex B (normative-optional, web required) specification text came up in a few topics. There's disagreement about the current role and appropriateness, to be revisited in a future meeting.
1. Several needs-consensus pull requests reached consensus, tightening up semantics in edge cases.
1. ES2017 was officially approved by the committee to be sent to the ECMA General Assembly for approval.

**Comment by leobalter** [comment]
I'm updating this tomorrow in the morning to add the summary. (Away from a
computer rn, phone only)

One note: the CoC draft will be sent to the ECMA Executive Committee for
review and approval through vote.

I wrote a medium post about it as well, I can include the link tomorrow as
I'll publish it along.

**Comment by littledan** [comment]
Something that would be really nice in these notes would be cross-references to both the slides and the repositories for each proposal.

**Comment by leobalter** [comment]
@littledan @rwaldron I pushed a new commit with the summary including not only the link to the proposals but also the respective notes.

File: es8/2017-03/mar-21.md [hunk 32]
```diff
@@ -0,0 +1,1467 @@
...
+
+DH: Our sound person says you are showing up as self-muted.
+
+CP: ...
+
+DFV: I'd say you should move on and try again.
+
+AWB:  We'll go ahead and move on to the 15-minute items.
+
+## 10.i.a. Template Literal Updates
```
Comment (bnoordhuis) [inline es8/2017-03/mar-21.md]: The sections would benefit from linking to the proposals.  For example, I glean this section is about https://tc39.github.io/proposal-template-literal-revision/ but it's hard to say for sure.

Comment (rwaldron) [inline es8/2017-03/mar-21.md]: Fixed in master. Normally these are added to each topic as the notes are taken

**Comment by leobalter** [comment]
Thanks @rwaldron and everyone who helped! +1 to the collaborative work.

**Pull request merged**